src/ngraph/ngraph_emitter.cc

```diff
@@ -121,8 +123,8 @@ NgraphNodePtr Emitter::ReduceAxes(
 NgraphNodePtr Emitter::ReduceAxes(
     const NodePtr& node,
     const std::function<NgraphNodePtr(const NgraphNodePtr&,
-                                      const ngraph::AxisSet&)>& func) {
-  auto input = op_map_[node->inputs_[0]];
+                                      const ngraph::AxisSet&)>& func) const {
+  auto input = checked_lookup(op_map_, node->inputs_[0]);
```

Comment:
Why not .at()? This seems unnecessary for a safe lookup.

Reply:
What i wanted here was the ability to make the method `const`.  

`operator[]` on `std::map` is not `const`, because it will create an element if one doesn't exist.  So I created `checked_lookup` as a convenience method to do the lookup without the possibility of modifying the map.

I defined `checked_lookup(...)` to fail loudly (exception, I think) it the lookup key is not present in the map, for several reasons:
- I wanted to keep its signature simple, s.t. the return value is the key's dependent value.
- I didn't want to introduce a "maybe" or "optional" type to represent the possibility of a failed lookup, because in this context a failed lookup really would be a surprise.

Reply:
Does at() not meet these requirements? Just curious, not against the utility.

Reply:
Actually yeah, it does. Good catch.  I thought you were asking about using `at()` on the vector that's present on that line.  Will fix.